### ISSUE TITLE
Weirdness in RUN / shell substitution

### ISSUE THREAD

**Issue by dieswaytoofast** [seatgeek/docker-build-cacher#3]
Use the following Dockerfile
```
FROM busybox
RUN echo "foo    bar"
```
(note - multiple spaces between "foo" and "bar"

output of the RUN from `docker build` is
```
Step 2/2 : RUN echo "foo    bar"
 ---> Running in 849afc43fcf9
foo    bar
```
output of the run from `docker-build-cacher` is
```
Step 2/2 : RUN echo "foo bar"
 ---> Running in 477b6b24a110
foo bar
```
(note, the spaces got squashed)

(context - this plays holy hell with doing checksums with `sha256sum` - the double spaces get squashed...)

**Comment by lorenzo** [comment]
True, the parser does strip extra whites off in shells. Is this something you rely on?

**Comment by dieswaytoofast** [comment]
Yeah, a fairly common pattern in Dockerfiles when one needs to validate a file is doing something like
`echo "$SOME_CHECKSUM  FILE_TO_BE_CHECKSUMMED | sha256sum -c - `
the thing is, `sha256sum` expects *two* spaces between the checksum and the filename 🙄

e.g. https://github.com/erlang/docker-erlang-otp/blob/3765be4573e249004e48a5de9c94127a214b1e6f/22/Dockerfile#L20

**Comment by lorenzo** [comment]
Interesting, I really don't understand why that's needed, I'll look into changing the parser to preserve the whitespace

**Comment by dieswaytoofast** [comment]
When downloading a file from a remote source, and validation is necessary, you do it by validating the checksum.
You do that by running a checksum on the file, and then compare it with the value you already know.
You can do *that* by either doing a bunch of shell gymnastics ( _run sha256sum on the file. if the returned value equals what i already know, then yay. else fial_ )
or
you can do it all on one line using `echo "$SOME_CHECKSUM FILE_TO_BE_CHECKSUMMED | sha256sum -c -`

😜
